### PR TITLE
Improve position:fixed support in layout_2020

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #___ (GitHub issue number if applicable)

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because ___

### CODE REVIEW

**Review by SimonSapin — APPROVED** [review]

File: components/layout_2020/display_list/mod.rs [hunk 1]
```diff
@@ -301,6 +267,67 @@ impl<'a> BuilderForBoxFragment<'a> {
...
+        }
+
+        // TODO(mrobinson): Eventually this should use the spatial id of the reference
+        // frame that is the parent of this one once we have full support for stacking
+        // contexts and transforms.
+        let spatial_id = wr::SpatialId::root_reference_frame(builder.wr.pipeline_id);
+        builder.current_space_and_clip = wr::SpaceAndClipInfo {
+            spatial_id,
+            clip_id: builder.current_space_and_clip.clip_id,
+        };
```
[the remark this answers is not in the document]
Comment (mrobinson) [reply components/layout_2020/display_list/mod.rs]: Yep. I'm happy to integrate this change.

[the remark this answers is not in the document]
Comment (mrobinson) [reply components/layout_2020/display_list/mod.rs]: Okay. I've pushed a new version of the branch that also does away with the temporary variable.

**Comment by SimonSapin** [comment]
@bors-servo retry

```
      > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/sdk-common/26.1.3/sdk-common-26.1.3.jar'.
         > Connection reset
```

**Pull request merged**